x/payment/payfordata.go

```diff
@@ -61,6 +62,10 @@ func BuildPayForData(
 	message []byte,
 	gasLim uint64,
 ) (*types.MsgWirePayForData, error) {
+	// sanity check nID size
+	if len(nID) != types.NamespaceIDSize {
+		return nil, fmt.Errorf("expected namespace ID of size %d, got %d", types.NamespaceIDSize, len(nID))
+	}
```

Comment:
this is a good check, but do you think it makes sense to add it to the `NewWirePayForData` function? Perhaps that would be less explicit while reading code, but then it performs the same check each time we create a wirePFD.

better yet, we could just call `ValidateBasic` before returning from `NewWirePayForData`, which performs this check along with a few others. This function has to pass if the wirePFD will make it into the mempool, so its probably good to run ahead of time anywho.

Reply:
https://github.com/celestiaorg/celestia-app/pull/418/commits/9ed560c46e791aec558ef66d4e44610b4dc3f2b1